src/identity/app.rs

```diff
@@ -99,6 +100,19 @@ impl PublicId {
     pub fn owner(&self) -> &ClientPublicId {
         &self.owner
     }
+
+    /// Returns the PublicId serialised and encoded in base64.
+    pub fn encode_to_base64(&self) -> String {
+        let pub_id = unwrap!(bincode::serialize(&self));
+        base64::encode(&pub_id)
+    }
+
+    /// Create from base64 encoded string.
+    pub fn decode_from_base64(encoded: &str) -> Result<Self, Error> {
```

Comment:
I think we should make the method's signature:
```rust
pub fn decode_from_base64<T: ?Sized + AsRef<[u8]>>(encoded: &T) -> 
```
since this matches `base64::decode`'s input and is a bit more user-friendly.